### PR TITLE
Manual.md: document Transitional packages

### PR DESCRIPTION
**Pull request by maxice8**
I saw 2 implementations of transitional packages.

1. old package stays but is turned into meta-package that points to new package ( lemonbuddy -> polybar )
2. old package is removed, new subpackage is created with name of old package that points to n ew package ( gnome-tweak-tool -> gnome-tweaks, upower0 -> upower )

i documented regarding 2, also having different implementations of such trivialities is annoying.

### CODE REVIEW

**Comment by maxice8** [comment]
i am also not sure if to call them `transitive` or `dummy` or `transitive dummy`

**Comment by Hoshpak** [comment]
I would call them transitional packages as I don't recognize any transitivity there.

**Comment by maxice8** [comment]
On Fri, Mar 30, 2018 at 10:06:07AM +0000, Helmut Pozimski wrote:
> I would call them transitional packages as I don't recognize any transitivity
> there.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.*
> 

yeah, i updated the PR

**Comment by Gottox** [comment]
@voidlinux/pkg-committers I like this approach. comments?

File: Manual.md [hunk 1]
```diff
@@ -1227,6 +1228,16 @@ The following variables influence how Haskell packages are built:
 - `make_build_args`: This is passed as-is to `stack build ...`, so
   you can add your `--flag ...` parameters there.
 
+<a id="pkgs_transitional"></a>
+### Transitional packages
+
+Transitional packages are used to move users into a new package. It is commonly
+used when upstream changes it's name and Void Linux changes the package name to
+reflect such change. A transitive package must be an empty subpackage of the
```
Comment (ebfe) [inline Manual.md]: s/transitive/transitional/

File: Manual.md [hunk 2]
```diff
@@ -1227,6 +1228,16 @@ The following variables influence how Haskell packages are built:
...
+package of the new main package, on the subpackage, it must have `noarch=yes`, 
+`short_desc+"= - transitional package"` and
+`depends="${pkgname}-${version}_${revision}"`
```
Comment (ebfe) [inline Manual.md]: s/pkgname/sourcepkg/

**Comment by maxice8** [comment]
Thanks for catching, will fix when I get home in about 1 hour

**Comment by maxice8** [comment]
Will fix whenever I get home eta 1h

**Comment by ebfe** [comment]
Maybe it would be good to indentify/explain why these are needed at all. Why doesn't replaces=(/provides=) work? Also see #11804.

These transitional packages have the problem that you never know when they can be safely removed, so i'm not sure if they should be encouraged.

File: Manual.md [hunk 3]
```diff
@@ -1227,6 +1228,16 @@ The following variables influence how Haskell packages are built:
 - `make_build_args`: This is passed as-is to `stack build ...`, so
   you can add your `--flag ...` parameters there.
 
+<a id="pkgs_transitional"></a>
+### Transitional packages
+
+Transitional packages are used to move users into a new package. It is commonly
+used when upstream changes it's name and Void Linux changes the package name to
```
Comment (newbluemoon) [inline Manual.md]: I think it must be “its name”.

**Comment by maxice8** [comment]
As far as I can tell  replaces provides doesn't update people with the old
package

File: Manual.md [hunk 4]
```diff
@@ -1227,6 +1228,16 @@ The following variables influence how Haskell packages are built:
...
+reflect such change. A transitive package must be an empty subpackage of the
+package of the new main package, on the subpackage, it must have `noarch=yes`,
```
Comment (newbluemoon) [inline Manual.md]: Is there a “package” too much? And maybe split into two sentences for better readability?

**Comment by ebfe** [comment]
> As far as I can tell replaces provides doesn't update people with the old package

But that's exactly what replaces is for. replaces=oldname>=0 and revbumping all revdeps so they refer to the new name should take care of that. As far as i understand it the only reason to use transitional packages is to avoid the revbump of the revdeps. But all in all replaces should be the "default" rename strategy as it also takes care of removing the old package from the system and doesn't clobber the repo with dummy packages.

**Comment by maxice8** [comment]
oh, i wasn't aware of that (no docs), would be really good if this knowledge was added to the docs.

**Comment by maxice8** [comment]
@ebfe using replaces= also require changes to the package that is going to be removed ?

**Comment by Duncaen** [comment]
Transitional packages are good to replace manually installed packages without rdevs.
If nothing pulls in a package into the transaction it would never be considered for `replaces=`.
The problem with them is that transitional packages would replace the manually installed package and install the package its replaced by as automatically installed.

**Comment by ebfe** [comment]
> If nothing pulls in a package into the transaction it would never be considered for replaces=.

If that's the case then maybe that should be fixed instead of adding transitional packages.

**Comment by ebfe** [comment]
> > If nothing pulls in a package into the transaction it would never be considered for replaces=.
> 
> If that's the case then maybe that should be fixed instead of adding transitional packages.

Which of course isn't trivial to do since it's used in combination with
provides for virtual packages which relies on this
behaviour. And is at the same time broken by it ((releated bug
https://github.com/voidlinux/xbps/issues/257). Which leads us back to #11804.

File: Manual.md [hunk 5]
```diff
@@ -1227,6 +1228,16 @@ The following variables influence how Haskell packages are built:
 - `make_build_args`: This is passed as-is to `stack build ...`, so
   you can add your `--flag ...` parameters there.
 
+<a id="pkgs_transitional"></a>
+### Transitional packages
+
+Transitional packages are used to move users into a new package. It is commonly
+used when upstream changes it's name and Void Linux changes the name to reflect
```
Comment (pullmoll) [inline Manual.md]: s/it's/its/

Comment (maxice8) [reply Manual.md]: fixed,, thanks

**Comment by maxice8** [comment]
soooo, what is the current plan ? document replaces= / provides= and make Transitional packages documentation just explain how to use those to make a transitional package ?

**Comment by ebfe** [comment]
I think it became clear that renaming packages isn't a 'triviality' so I propose:

```markdown
# Renaming packages

Don't do it unless you know what you are doing.
```

until replaces/provides semantics are clear and implemented in a sane way in xbps. (only half joking)

**Comment by maxice8** [comment]
> # Renaming packages
>
> Don't do it unless you know what you are doing.

Nobody does.

I'll shelf this until xbps people sort it out.

**Pull request closed without merging**